### ISSUE TITLE
Is it cache the false ACL result

### ISSUE THREAD

**Issue by eydonmlau** [iegomez/mosquitto-go-auth#278]
Hi, I use the cache setting like: 
```
auth_opt_cache true
auth_opt_cache_type go-cache
auth_opt_cache_reset true
auth_opt_cache_refresh true
```
When ACL check is not pass, it would cache all the same like the pass way, Is it correct?
I checked the code, found the code like this:
<img width="703" alt="checkACL" src="https://github.com/iegomez/mosquitto-go-auth/assets/19401578/9a9f66bf-859a-4200-b1ed-83246b7406ef">
    You can imagine a scenario, The first time a gateway(mqtt client) didn't add to application platform, and the acl would not ok first, then the gateway is added to the application platform by user, but acl also would not pass because of the cache. And if the flag `auth_opt_cache_refresh` was setted, it would never  be ok. I feel that this logic is not in line with practical applications. 
    In this case, what should I do ? may be set the flag `auth_opt_cache_refresh=false` is the only thing to prevent this case?
best wishes~

**Comment by iegomez** [comment]
Yeah, I think you need to disable refreshment altogether or handle it in the connecting side to do an exponential backoff to break the cache seconds you set in order to suit your use case.
If you have a better idea, I'm all ears.

Cheers!
